src/dmd/expression.d

```diff
@@ -3635,6 +3635,23 @@ extern (C++) final class ArrayLiteralExp : Expression
     }
 }
 
+/***********************************************************
+ * verify that TypeInfo for the expression type exists or generate it
+ * if necessary, i.e. the current scope is not during CTFE, but
+ * the type info will be needed by the code generation later.
+ *
+ * Params:
+ *        ae = the array literal to verify
+ *        sc = the scope in which the type info should be analyzed. If null,
+ *             the global scope of Module.rootModule is used
+ */
+void verifyTypeInfo(ArrayLiteralExp ae, Scope* sc)
```

Comment:
Is there any particular reason why these `verifyTypeInfo` functions are module-level functions?  Would it be more appropriate for them to be member functions of `ArrayLiteralExp`, `AssocArrayLiteralExp`, etc...?

Reply:
Not a particular reason, I've just been trying to follow the trend to keep members in AST classes to a minimum.
I just notice that verifyTypeInfo is never called for CatEp and EqualExp, that must have been gone during the rebase (maybe because semanticTypeInfo is now already being called). I'll try to figure out whether they are obsolete now...